service-python/main_consumer.py

```diff
@@ -37,24 +39,33 @@ def _create_connection(self):
             except Exception:
                 logging.warning(f"RabbitMQ Connection Failed. Retrying in 30s ({i + 1}/{self.config['retry_limit']})")
                 sleep(30)
+        return None
 
     def setup_queues(self):
         channel = self.connection.channel()
         queue_setup(channel)
         self.channel = channel
 
 
+# This file will get copied to the docker image's root folder(src) when being built
+# When run, it attempts to create a pika.BlockingConnection() with the settings in CONSUMER_CONFIG
+# There are 2 retry levels for the consumer. The first being in _create_connection() which is based on retry_limit
+# The other being when the app crashes which is based on timeout
+# If it fails, it will try to delete any existing connection and the reference to the instantiated class
+# since the retry loop will recreate it
 if __name__ == "__main__":
 
     start_timer = None
     current_timer = None
 
     while(True):
+        consumer = None
         try:
             consumer = RabbitMQConsumer(CONSUMER_CONFIG)
-
-            consumer.channel.start_consuming()
+            if consumer.channel:
+                consumer.channel.start_consuming()
         except Exception:
+            del consumer
```

Comment:
Please put a logging warning here for the exception.  Looks like logging.warning has a exc_info parameter where you can specify the exception instance.